src/modules/simulation/gz_bridge/GZMixingInterfaceServo.cpp

```diff
@@ -33,6 +33,156 @@
 
 #include "GZMixingInterfaceServo.hpp"
 
+float
+GZMixingInterfaceServo::get_tilt_angle_max(const size_t index)
+{
+	switch (index) {
+	case 0: return _tilt_max_angle_0.get();
+
+	case 1: return _tilt_max_angle_1.get();
+
+	case 2: return _tilt_max_angle_2.get();
+
+	case 3: return _tilt_max_angle_3.get();
+
+	default: return NAN;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_tilt_angle_min(const size_t index)
+{
+	switch (index) {
+	case 0: return _tilt_min_angle_0.get();
+
+	case 1: return _tilt_min_angle_1.get();
+
+	case 2: return _tilt_min_angle_2.get();
+
+	case 3: return _tilt_min_angle_3.get();
+
+	default: return NAN;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_servo_angle_max(const size_t index)
+{
+	switch (index) {
+	case 0: return _servo_max_angle_0.get();
+
+	case 1: return _servo_max_angle_1.get();
+
+	case 2: return _servo_max_angle_2.get();
+
+	case 3: return _servo_max_angle_3.get();
+
+	case 4: return _servo_max_angle_4.get();
+
+	case 5: return _servo_max_angle_5.get();
+
+	case 6: return _servo_max_angle_6.get();
+
+	case 7: return _servo_max_angle_7.get();
+
+	default: return NAN;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_servo_angle_min(const size_t index)
+{
+	switch (index) {
+	case 0: return _servo_min_angle_0.get();
+
+	case 1: return _servo_min_angle_1.get();
+
+	case 2: return _servo_min_angle_2.get();
+
+	case 3: return _servo_min_angle_3.get();
+
+	case 4: return _servo_min_angle_4.get();
+
+	case 5: return _servo_min_angle_5.get();
+
+	case 6: return _servo_min_angle_6.get();
+
+	case 7: return _servo_min_angle_7.get();
+
+	default: return NAN;
+	}
+}
+
+int GZMixingInterfaceServo::get_active_output_count()
+{
+	return _control_surface_count.get() + _tilt_count.get();
+}
+
+bool GZMixingInterfaceServo::is_tiltrotor(const int index)
+{
+	if (_tilt_count.get() < 1) {
+		return false;
+	}
+
+	return (index > _control_surface_count.get() - 1) && (index < get_active_output_count());
+}
```

Comment:
I would try to remove vehicle specific handling in the gzbridge.

Reply:
done!